app/js/app/directives/graph_sketcher/GraphSketcher.js

```diff
@@ -0,0 +1,2586 @@
+                                        if (intpts[t].ind > tempMax.ind) {
+                                            upperRng.push(intpts[t]); 
+                                            intpts.pop(intpts[t]);
+                                        } else if (intpts[t].ind <= tempMax.ind && intpts[t].ind >= m.maxima[clickedKnotId].ind) {
+                                            tempCurveHigher.push(intpts[t]);
+                                            intpts.pop(intpts[t]);
+                                        } else if (intpts[t].ind <= m.maxima[clickedKnotId].ind && intpts[t].ind >= tempMin.ind) {
+                                            tempCurveLower.push(intpts[t]);
+                                            intpts.pop(intpts[t]);
+                                        } else if (intpts[t].ind < tempMin.ind) {
+                                            lowerRng.push(intpts[t]);
+                                            intpts.pop(intpts[t]);
+                                        }              
+                                    }
+
+                                    lowerRng.sort(function(a, b){return a.ind - b.ind});
+                                    upperRng.sort(function(a, b){return a.ind - b.ind});
+                                    tempCurveLower.sort(function(a, b){return a.ind - b.ind});
+                                    tempCurveHigher.sort(function(a, b){return a.ind - b.ind});
+                                    tempCurveL.pts = tempCurveLower;
+                                    tempCurveH.pts = tempCurveHigher;
+
+                                    // we have now split the curve into lowerRng and upperRng, plus tempCurveLower and tempCurveHigher
+                                    var lorx = m.maxima[clickedKnotId].x - tempMin.x;
+                                    var lory = m.maxima[clickedKnotId].y - tempMin.y;
+                                    var rorx = tempMax.x - m.maxima[clickedKnotId].x;
+                                    var rory = m.maxima[clickedKnotId].y - tempMax.y;
+                                    var dx = current.x - prevMousePt.x;
+                                    var dy = current.y - prevMousePt.y;
+                                    prevMousePt = current;
+                                    m.maxima[clickedKnotId].x += dx;
+                                    m.maxima[clickedKnotId].y += dy;
+
+                                    var lnrx = m.maxima[clickedKnotId].x - tempMin.x;
+                                    var lnry = m.maxima[clickedKnotId].y - tempMin.y;
+                                    var rnrx = tempMax.x - m.maxima[clickedKnotId].x;
+                                    var rnry = m.maxima[clickedKnotId].y - tempMax.y;
+                                    stretchMode = 2;
+                                    switch (stretchMode) {
+                                  
```

Comment:
All these magic numbers could do with refactoring, but perhaps not now . . .

Reply:
Agreed!